ambry-named-mysql/src/main/java/com/github/ambry/named/MySqlNamedBlobDb.java

```diff
@@ -156,157 +191,70 @@ public CompletableFuture<NamedBlobRecord> get(String accountName, String contain
     TransactionStateTracker transactionStateTracker =
         new GetTransactionStateTracker(remoteDatacenters, localDatacenter);
     return executeTransactionAsync(accountName, containerName, true, (accountId, containerId, connection) -> {
-      try (PreparedStatement statement = connection.prepareStatement(GET_QUERY)) {
-        statement.setInt(1, accountId);
-        statement.setInt(2, containerId);
-        statement.setString(3, blobName);
-        try (ResultSet resultSet = statement.executeQuery()) {
-          if (!resultSet.next()) {
-            throw buildException("GET: Blob not found", RestServiceErrorCode.NotFound, accountName, containerName,
-                blobName);
-          }
-          String blobId = Base64.encodeBase64URLSafeString(resultSet.getBytes(1));
-          Timestamp expirationTime = resultSet.getTimestamp(2);
-          Timestamp deletionTime = resultSet.getTimestamp(3);
-          long currentTime = System.currentTimeMillis();
-          if (compareTimestamp(expirationTime, currentTime) <= 0 && !includeExpiredOptions.contains(option)) {
-            throw buildException("GET: Blob expired", RestServiceErrorCode.Deleted, accountName, containerName,
-                blobName);
-          } else if (compareTimestamp(deletionTime, currentTime) <= 0 && !includeDeletedOptions.contains(option)) {
-            throw buildException("GET: Blob deleted", RestServiceErrorCode.Deleted, accountName, containerName,
-                blobName);
-          } else {
-            return new NamedBlobRecord(accountName, containerName, blobName, blobId, timestampToMs(expirationTime));
-          }
-        }
+      NamedBlobRecord record, recordV2 = null;
+      record = run_get(accountName, containerName, blobName, option, accountId, containerId, connection);
+      try {
+        recordV2 = run_get_v2(accountName, containerName, blobName, option, accountId, containerId, connection);
+      } catch (Exception e) {
+        logger.error("NAMED GET: Data Error: old table succeed while new table failed: ", e);
```

Comment:
this doesn't seems right. There will be a period of time when v1 table has more data than v2 table, some blobname would return a valid record in v1 but not_found in v2. and this shouldn't be an issue. 

we should probably do something like this to cover this case
```
record = run_get();
try {
    recordV2 = run_get_v2();
    if (!record.equalsWithoutVersion(recordV2)) {
        // log error
        // metrics
    }
} catch (Exception e) {
    if (isNotFoundException(e)) {
        // record metric here
    }
    // log an error 
    // record metric here
}
```

Reply:
Right, but there won't be too many such cases, and this is just for logging/metrics purpose, I don't think we are aiming for perfect metrics (eg. 0), we are good if the metrics is small enough?

Reply:
We only have to run sync old to new table once after deployment. My idea is
1. Deploy the code that writes to both table 
2. Copy data from old table to new 

In this way, we will guarantee all the old data would be synced to new. 

However, we don't really know when we are going to do the step 2 so after step 1 there will be lots of missing data in new table. 

That's my plan here.

Reply:
I see, just updated in commit [359fb6f](https://github.com/linkedin/ambry/pull/2143/commits/359fb6f29d7756fe4bb290255c553791128f0142). Can you take another look?